core/templates/pages/contributor-dashboard-admin-page/contributor-admin-dashboard-page.component.ts

```diff
@@ -84,30 +134,131 @@ export class ContributorAdminDashboardPageComponent implements OnInit {
       this.isTranslationCoordinator = userInfo.isTranslationCoordinator();
 
       if (this.isTranslationCoordinator) {
-        this.CONTRIBUTION_TYPES = [this.TAB_NAME_TRANSLATION_SUBMITTER,
-          this.TAB_NAME_TRANSLATION_REVIEWER];
+        this.CONTRIBUTION_TYPES.push(
+          this.TAB_NAME_TRANSLATION_SUBMITTER,
+          this.TAB_NAME_TRANSLATION_REVIEWER);
+        this.contributorDashboardAdminStatsBackendApiService
+          .fetchAssignedLanguageIds(username).then(
+            (response) => {
+              this.assignedLanguageIds = response;
+              this.languages = this.languages.filter((
+                  languageItem) =>
+                this.assignedLanguageIds.includes(languageItem.id));
+              this.selectedLanguage = this.languages[0].language;
+              this.selectedLanguageId = this.languages[0].id;
+              this.translationReviewersCount = (
+                this.translationReviewersCountByLanguage[
+                  this.selectedLanguageId]);
+            });
       }
       if (this.isQuestionCoordinator) {
-        this.CONTRIBUTION_TYPES = [this.TAB_NAME_QUESTION_SUBMITTER,
-          this.TAB_NAME_QUESTION_REVIEWER];
-      }
-      if (this.isQuestionCoordinator && this.isTranslationCoordinator) {
-        this.CONTRIBUTION_TYPES = [this.TAB_NAME_TRANSLATION_SUBMITTER,
-          this.TAB_NAME_TRANSLATION_REVIEWER,
+        this.CONTRIBUTION_TYPES.push(
           this.TAB_NAME_QUESTION_SUBMITTER,
-          this.TAB_NAME_QUESTION_REVIEWER];
+          this.TAB_NAME_QUESTION_REVIEWER);
       }
 
+      this.filter = new ContributorAdminDashboardFilter(
+        [],
+        this.selectedLanguageId,
+        null,
+        this.selectedLastActivity);
+
       this.updateSelectedContributionType(this.CONTRIBUTION_TYPES[0]);
+
+      this.loadingMessage = '';
       this.changeDetectorRef.detectChanges();
     });
 
+    this.lastActivty = [0, 7, 30, 90];
+
     this.contributorDashboardAdminStatsBackendApiService
-      .fetchCommunityStats().then(
+      .fetchTopicsData().then(
         (response) => {
-          this.translationReviewersCount = response.translation_reviewers_count;
-          this.questionReviewersCount = response.question_reviewers_count;
-        });
+          this.topics = response;
+          response.map((
+              topic) =>
+            this.allTopicNames.push(topic.topic));
+          this.applyTopicFilter();
+        }
+      );
+  }
+
+  toggleLanguageDropdown(): void {
+    this.languageDropdownShown = !this.languageDropdownShown;
+  }
+
+  toggleActivityDropdown(): void {
+    this.activityDropdownShown = !this.activityDropdownShown;
+  }
+
+  applyTopicFilter(): void {
+    this.selectedTopicsIds = this.selectedTopics.map((
+        selectedTopic) => {
+      const matchingTopic = this.topics.find(
+        (topicChoice) => topicChoice.topic === selectedTopic);
```

Comment:
```suggestion
        topicChoice => topicChoice.topic === selectedTopic);
```
Nit: Don't think the parentheses are necessary if there's only one param. Ditto elsewhere.

Reply:
Fixed. everywhere.